http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -39,23 +38,59 @@
 import org.apache.commons.csv.CSVRecord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/hiscore")
 public class HiscoreService
 {
 	private static final Logger logger = LoggerFactory.getLogger(HiscoreService.class);
 
-	private static final HttpUrl RUNESCAPE_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_NORMAL_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_IRONMAN_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool_ironman/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_HARDCORE_IRONMAN_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool_hardcore_ironman/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_ULTIMATE_IRONMAN_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool_ultimate/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_DEADMAN_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool_deadman/index_lite.ws");
+	private static final HttpUrl RUNESCAPE_SEASONAL_DEADMAN_HISCORE_SERVICE = HttpUrl.parse("http://services.runescape.com/m=hiscore_oldschool_seasonal/index_lite.ws");
 
-	private HttpUrl url = RUNESCAPE_HISCORE_SERVICE;
+	private HttpUrl testUrl;
 
-	private HiscoreResultBuilder lookupUsername(String username) throws IOException
+	private HiscoreResultBuilder lookupUsername(String username, HiscoreEndpoint endpoint) throws IOException
 	{
+		HttpUrl url;
+
+		// I don't like this, there must be a better way to set the URL for the test
```

Comment:
Why not make the URL a parameter?

Reply:
Since the only use case for setting the Hiscore URL to an arbitrary value is the unit test for this class, is it worthwhile to add a method like that? I'm not super familiar with unit testing but in my head the best thing to do would be to expose a method to only the test class so that anyone trying to use the real functionality doesn't try to set the URL manually. This is pretty much what was there before since I wasn't sure how to fix it.

Reply:
Maybe it would be best to subclass the service in the test and override a getUrl(HiscoreEndpoint endpoint)?